artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/TypedProperties.java

```diff
@@ -530,6 +547,10 @@ public synchronized String toString() {
    // Private ------------------------------------------------------------------------------------
 
    private synchronized void doPutValue(final SimpleString key, final PropertyValue value) {
+      if (internalPropertyPredicate != null && internalPropertyPredicate.test(key)) {
```

Comment:
No need to check if `internalProperties = true`
```java
if (!internalProperties && internalPropertyPredicate != null && internalPropertyPredicate.test(key)) {
```

Reply:
And what about `doRemoveProperty`?
I would use an `int internalProperties` instead of a `boolean` to make possible on `doRemoveProperty` to decrease the count and avoid performing cleanup of internal properties when there aren't anymore, wdyt?
The original version was missing a way to keep that flag consistent...

Reply:
agree on this check: internalProperties for the put.

dis-agree on making an int, this flag is ONLY used for the cleanup to avoid us cycling.

Reply:
@michaelandrepearce Bud I have just sent a comment as if I've been you!! :O

Reply:
@franz1981 seems GitHub has a very poor ACL system.... glad my org doesnt use it!!

Anyhow back to case, whilst yes that maybe the case if you remove, at the end of the day, its a performance hint, to avoid the cycle. If we happen to cycle still, then functionality wise we are not incorrect, just a performance annoyance.  (which without this, as you have currently without this, would be occuring all the time)

Reply:
Nice, so it is fine :+1: 
Yep I will ping the others of my group about this weird behaviour of github :O